src/legacy/state/orders/middleware/appziMiddleware.ts

```diff
@@ -44,8 +44,12 @@ function _triggerNps(
   const openSince = order?.openSince
   const explorerUrl = getExplorerOrderLink(chainId, orderId)
 
-  // Open Appzi NPS for limit orders only if they were filled before `PENDING_TOO_LONG_TIME` since creating
-  if (order?.class === OrderClass.LIMIT && npsParams?.traded && isOrderInPendingTooLong(openSince)) {
+  if (
+    // Do not show NPS if the order is hidden and expired
+    (order?.isHidden && npsParams?.expired) ||
+    // Open Appzi NPS for limit orders only if they were filled before `PENDING_TOO_LONG_TIME` since creating
+    (order?.class === OrderClass.LIMIT && npsParams?.traded && isOrderInPendingTooLong(openSince))
+  ) {
     return
   }
```

Comment:
```suggestion
  // Open Appzi NPS for limit orders only if they were filled before `PENDING_TOO_LONG_TIME` since creating
  const isLimitOrderRecentlyTraded =
    order?.class === OrderClass.LIMIT && npsParams?.traded && isOrderInPendingTooLong(openSince)
  
  // Do not show NPS if the order is hidden and expired
  if ((order?.isHidden && npsParams?.expired) || isLimitOrderRecentlyTraded) {
    return
  }
```